common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -1318,7 +1318,9 @@ bool mon_wlan_hal_dwpal::process_dwpal_nl_event(struct nl_msg *msg)
     nla_parse(tb, NL80211_ATTR_MAX, genlmsg_attrdata(gnlh, 0), genlmsg_attrlen(gnlh, 0), NULL);
 
     if (tb[NL80211_ATTR_IFINDEX]) {
-        if_indextoname(nla_get_u32(tb[NL80211_ATTR_IFINDEX]), ifname);
+        auto index            = nla_get_u32(tb[NL80211_ATTR_IFINDEX]);
+        auto iface_index_name = beerocks::net::network_utils::linux_get_iface_name(index);
+        std::copy_n(iface_index_name.begin(), iface_index_name.length(), ifname);
```

Comment:
Here also, declare `ifname` to be a `std::string` instead of `char[]`, and on each comparation to `ifname` use `==` operator, instead of `.compare()`.

Reply:
fixed